File: src/main/java/com/fendyk/clients/fetch/FetchMinecraftUser.java
package com.fendyk.clients.fetch;

import com.fendyk.DTOs.MinecraftUserDTO;
import com.fendyk.DTOs.updates.UpdateMinecraftUserDTO;
import com.fendyk.Main;
import com.fendyk.clients.FetchAPI;
import okhttp3.Request;
import okhttp3.RequestBody;

import java.io.IOException;
import java.util.UUID;

public class FetchMinecraftUser extends FetchAPI<MinecraftUserDTO, UpdateMinecraftUserDTO> {
    public FetchMinecraftUser(String url) {
        super(url, MinecraftUserDTO.class);
    }
}

File: src/main/java/com/fendyk/clients/fetch/FetchMinecraftUser.kt
package com.fendyk.clients.fetch

import com.fendyk.DTOs.MinecraftUserDTO
import com.fendyk.DTOs.updates.UpdateActivitiesDTO
import com.fendyk.DTOs.updates.UpdateMinecraftUserDTO
import com.fendyk.clients.FetchAPI
import java.util.*
import java.util.concurrent.CompletableFuture

class FetchMinecraftUser : FetchAPI<UUID, MinecraftUserDTO, UpdateMinecraftUserDTO>(MinecraftUserDTO::class.java) {
    override fun get(key: UUID): CompletableFuture<MinecraftUserDTO> {
        return fetch("/minecraftusers/$key", RequestMethod.GET, null)
    }

    override fun update(key: UUID, data: UpdateMinecraftUserDTO): CompletableFuture<MinecraftUserDTO> {
        return fetch("/minecraftusers/$key", RequestMethod.PATCH, data)
    }
}
